### PR TITLE
Feature: simpler usage approach

### PR DESCRIPTION
**Pull request by pke**
instead of this code:

```jsx
<ScrollSync>
  <div style={{ display: 'flex', position: 'relative', height: 300 }}>
    <ScrollSyncPane>
      <div style={{overflow: 'auto'}}>
        <section style={{ height: 500 }}>
``` 

the component can now be used like this:

```jsx
<ScrollSync style={{ display: 'flex', position: 'relative', height: 300 }}>
  <section style={{ height: 500 }}>
``` 

It also no longer makes use of `context` and `ReactDOM.find`.

### CODE REVIEW

**Comment by pke** [comment]
@okonet I wonder, why do you choose to one time call `addEvents` inside `requestAnimation` frame and the other time don't`

File: src/ScrollSync.js [hunk 1]
```diff
@@ -81,6 +76,14 @@ export default class ScrollSync extends Component {
   }
 
   render() {
-    return React.Children.only(this.props.children)
+    return (<div style={{ overflow: 'hidden', ...this.props.style }}>
```
Comment (okonet) [inline src/ScrollSync.js]: What if I want to use it with something different than DOM or with a different tag? Also, I don't like it is inline styles only.

File: src/ScrollSync.js [hunk 2]
```diff
@@ -81,6 +76,14 @@ export default class ScrollSync extends Component {
...
+      {
+        React.Children.map(this.props.children, child => (
+          <ScrollSyncPane register={this.registerPane}>
```
Comment (okonet) [inline src/ScrollSync.js]: My approach allows you to wrap any element down the tree. This doesn't.

**Comment by okonet** [comment]
Although you removed some boilerplate, your approach is much more limited as mine. For example, you can't wrap arbitrary elements in the DOM tree anymore since you're creating the `ScrollSyncPane` as an immediate children of the container. Also, applying styles isn't something I want this library to be concerned about since it can be anything from inline styles to CSS or even CSS-in-JS.

**Pull request closed without merging**